__tests__/app.test.js

```diff
@@ -221,7 +220,257 @@ describe("POST /api/articles/:article_id/comments", () => {
+          .delete(`/api/comments/${commentId}`)
+          .expect(204);
+      })
+      .then(() => {
+        return request(app)
+          .get("/api/articles/1?comment_count=true")
+          .expect(200)})
+      .then(({ body }) => {
+        expect(body.article.comment_count).toBe(testedCommentCount);
+      });
+  });
+  test("DELETE 404: responds with error when comment_id does not exist", () => {
+    return request(app)
+      .delete("/api/comments/500")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Requested information not found!");
+      });
+  });
+  test("DELETE 400: responds with 'Bad Request!...' when comment_id entered in wrong format", () => {
+    return request(app)
+      .delete("/api/comments/wrong_format")
+      .expect(400)
+      .then(({ res }) => {
+        expect(res.statusMessage).toBe(
+          "Invalid request: not a valid integer entered!"
+        );
+      });
+  });
+});
+
+describe("GET /api/articles?queries", () => {
+  test("GET 200: array of article objects filtered by topic and sorted by date in descending order", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeInstanceOf(Array);
+        expect(articles).toHaveLength(12);
+        expect(articles[0].topic).toEqual("mitch");
+        expect(articles).toBeSorted({ key: articles.created_at })
+      });
+  });
+  test("GET 200: array of article objects sorted by a specified column and in the specified order", () => {
+    return request(app)
+      .get("/api/articles?sort_by=votes&order=asc")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeInstanceOf(Array);
+        expect(articles).toHaveLength(12);
+        expect(articles).toBeSorted({ key: articles.votes, descending: false })
+      });
+  });
+  test("GET 200: array of article objects filtered by topic, sorted by a specified column, and in the specified order", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch&sort_by=votes&order=asc")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeInstanceOf(Array);
+        expect(articles).toHaveLength(12);
+        expect(articles[0].topic).toEqual("mitch");
+        expect(articles).toBeSorted({ key: articles.votes, descending: false })
+      });
+  });
+  test("GET 200: array of article objects sorted by a default column and in the default order when no query is specified", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeInstanceOf(Array);
+        expect(articles).toHaveLength(12);
+        expect(articles).toBeSorted({ key: articles.created_at, descending: true })
+      });
+  });
+});
```

Comment:
There are some missing sad path tests for the queries: 

400 ?sort_by=bananas
400 ?order= asc&desc
200 valid topic but there are no articles associated. I think for NC news, "paper" has no corresponding articles (I might be wrong though)